bothan-core/src/manager/crypto_asset_info/manager.rs

```diff
@@ -188,11 +198,19 @@ impl<S: Store + 'static> CryptoAssetInfoManager<S> {
             .map_err(|e| SetRegistryError::InvalidRegistry(e.to_string()))?;
 
         self.store
-            .set_registry(registry, hash.to_string())
+            .set_registry(registry.clone(), hash)
             .await
             .map_err(|_| SetRegistryError::FailedToSetRegistry)?;
 
-        set_workers_query_ids(&self.workers, &self.store.get_registry().await).await;
+        // drop old workers to kill connection
+        let mut locked_workers = self.workers.lock().await;
+        *locked_workers = Vec::with_capacity(0);
+
+        // wait a bit for connections to clear up
+        sleep(Duration::from_secs(1)).await;
```

Comment:
After discuss with swit timeout 1 sec feel bad smell, we understand why 1 sec should enough but just add TODO in case we try to improve how we wait other workers cleanup before build the new one